### PR TITLE
Improving logging message, coverage

### PR DESCRIPTION
**Pull request by doncem**
Notes:
- [ ] Accumulated multi-project coverage report uploaded to Codacy
- [ ] Travis only runs against master and feature branches
- [x] Request log message improvement:
  - removed arbitrary markers containing request uri, time the response took. Giving this information in the message
  - leaving request method and response code. I doubt method is needed
- [x] Log format slightly adjusted:
  - adding date format
  - adding log level spacing
  - reducing namespace clog
  - adding line number

### CODE REVIEW

File: java-logging-spring/src/test/java/uk/gov/hmcts/reform/logging/spring/RequestLoggingComponentTest.java [hunk 1]
```diff
@@ -50,24 +52,41 @@ public void listenForEvents() throws JoranException, IOException {
     @Test
     public void requestProcessedMessageShouldBeLoggedForPublicResource() throws Exception {
         ResponseEntity<String> response = restTemplate.getForEntity("/public", String.class);
+
         assertThat(response.getBody()).isEqualTo("OK");
-        assertThat(loggedEvents()).extracting("message").containsOnlyOnce("Request processed");
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+
+        Thread.sleep(10);// how to fix this? test fails to see the log message otherwise
```
Comment (KarolJastrzebski) [inline java-logging-spring/src/test/java/uk/gov/hmcts/reform/logging/spring/RequestLoggingComponentTest.java]: Maybe waiting for 5 seconds until message gets logged, or fail? Have a look at http://tempusfugitlibrary.org/documentation/junit/waiting/

File: java-logging-spring/src/test/java/uk/gov/hmcts/reform/logging/spring/RequestLoggingComponentTest.java [hunk 2]
```diff
@@ -50,24 +52,41 @@ public void listenForEvents() throws JoranException, IOException {
     @Test
     public void requestProcessedMessageShouldBeLoggedForPublicResource() throws Exception {
         ResponseEntity<String> response = restTemplate.getForEntity("/public", String.class);
+
         assertThat(response.getBody()).isEqualTo("OK");
-        assertThat(loggedEvents()).extracting("message").containsOnlyOnce("Request processed");
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
```
Comment (KarolJastrzebski) [inline java-logging-spring/src/test/java/uk/gov/hmcts/reform/logging/spring/RequestLoggingComponentTest.java]: This seems unnecessary as the test is about logging a request regardless of the response's status.

File: java-logging-spring/src/test/java/uk/gov/hmcts/reform/logging/spring/RequestLoggingComponentTest.java [hunk 3]
```diff
@@ -50,24 +52,41 @@ public void listenForEvents() throws JoranException, IOException {
...
         // and get executed no matter request is allowed or not
         ResponseEntity<String> response = restTemplate.getForEntity("/protected", String.class);
+
         assertThat(response.getBody()).isNotEqualTo("OK");
-        assertThat(loggedEvents()).extracting("message").containsOnlyOnce("Request processed");
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.FORBIDDEN);
+
+        assertThat(loggedEvents().stream().filter(
+            event -> event.getFormattedMessage().startsWith("Request GET /protected processed")
+        ).collect(Collectors.toList())).size().isEqualTo(1);
```
Comment (KarolJastrzebski) [inline java-logging-spring/src/test/java/uk/gov/hmcts/reform/logging/spring/RequestLoggingComponentTest.java]: Why changing styles from assertj to hamcrest? This looks very verbose in comparison to the replaced approach.

File: src/main/java/uk/gov/hmcts/reform/logging/filters/RequestStatusLoggingFilter.java [hunk 4]
```diff
@@ -39,27 +39,36 @@ public void init(FilterConfig filterConfig) throws ServletException {
...
             throw e;
         }
     }
 
-    private LogstashMarker markersFor(ServletRequest request, ServletResponse response, long startTime) {
-        HttpServletRequest httpServletRequest = (HttpServletRequest) request;
+    private String buildLogMessage(String requestMethod, String requestUri, long period, boolean isSuccess) {
+        String status = isSuccess ? "processed" : "failed";
 
+        return String.format("Request %s %s %s in %dms", requestMethod, requestUri, status, period);
```
Comment (KarolJastrzebski) [inline src/main/java/uk/gov/hmcts/reform/logging/filters/RequestStatusLoggingFilter.java]: Why to relieve MessageFormatter of its duties? And how is it better than logger's implementation?

Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/logging/filters/RequestStatusLoggingFilter.java]: i would use message formatter but there is no such implementation for logger when you are passing the exception through

File: java-logging-spring/src/test/java/uk/gov/hmcts/reform/logging/spring/RequestLoggingComponentTest.java (hunk 3 above)
Comment (doncem) [reply java-logging-spring/src/test/java/uk/gov/hmcts/reform/logging/spring/RequestLoggingComponentTest.java]: could not regex match against the list of strings. the test case is now replaced with the previous suggestion where i await of logged event to happen

**Pull request merged**